src/workerd/api/http.c++

```diff
@@ -2048,6 +2049,36 @@ jsg::Promise<QueueResponse> Fetcher::queue(
   });
 }
 
+jsg::Promise<kj::Maybe<jsg::V8Ref<v8::Value>>> Fetcher::getRpcAndCallRemote(
+    jsg::Lock& js, kj::String method, jsg::Optional<kj::Array<jsg::Value>> args) {
+  auto& ioContext = IoContext::current();
+  // TODO(now): Just a reminder for the future, rename this operation name.
+  auto worker = getClient(ioContext, nullptr, "rpcTest"_kjc);
+  auto event = kj::refcounted<api::JsRpcServiceCustomEventImpl>(9);
+  auto eventRef = kj::addRef(*event);
+
+  return ioContext.awaitIo(js, worker->customEvent(kj::mv(eventRef)).attach(kj::mv(worker)),
+      [event=kj::mv(event), &ioContext, method=kj::mv(method), args=kj::mv(args)]
+      (jsg::Lock& js, WorkerInterface::CustomEvent::Result result) mutable {
+    // We read the rpc capability off the event to get around customEvent's return type limitation.
+    auto& client = KJ_REQUIRE_NONNULL(event->clientCap);
+    auto builder = client.callRequest();
+    builder.setMethodName(kj::mv(method));
+    // TODO(now): We should serialize the arguments we want to pass, and set them as well.
+    builder.initSerializedArgs().initV8Serialized(0);
+
+    return ioContext.awaitIo(js, builder.send(),
+          [](jsg::Lock& js, capnp::Response<rpc::JsRpcTarget::CallResults> result)
+              -> kj::Maybe<jsg::V8Ref<v8::Value>> {
+      auto serializedResult = result.getResult().getV8Serialized();
+      auto deserialized = jsg::deserializeV8Rpc(js, kj::heapArray(serializedResult.asBytes()));
+
+      v8::HandleScope scope(js.v8Isolate);
```

Comment:
Use `js.withinHandleScope(...)` here instead of using `v8::HandleScope` directly

Reply:
But the key question is whether the handlescope is needed here at all. We're already in a JS promise continuation at this point so we should already have a HandleScope on the stack.

Reply:
Yeah this looks unnecessary, there are no handles being created in this scope.